scripts/lmdk/build_module.py

```diff
@@ -0,0 +1,94 @@
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+
+MIN_PYTHON_VERSION = 3, 8
+assert sys.version_info >= MIN_PYTHON_VERSION, \
+	f"Python {MIN_PYTHON_VERSION} or above is required."
+
+# Constant value resolves SOF_TOP directory as: "this script directory/.."
+SOF_TOP = pathlib.Path(__file__).parents[1].resolve()
+west_top = pathlib.Path(SOF_TOP, "../..").resolve()
+LMDK_BUILD_DIR = west_top / "sof" / "lmdk"
+RIMAGE_BUILD_DIR = west_top / "build-rimage"
+
+if py_platform.system() == "Windows":
+	xtensa_tools_version_postfix = "-win32"
+elif py_platform.system() == "Linux":
+	xtensa_tools_version_postfix = "-linux"
+else:
+	xtensa_tools_version_postfix = "-unsupportedOS"
+	warnings.warn(f"Your operating system: {py_platform.system()} is not supported")
+
+
+class stores_libraries_arguments(argparse.Action):
+	"""Stores libraries arguments whether provided module name is supported."""
+	def __call__(self, parser, namespace, values, option_string=None):
+		setattr(namespace, "libraries", values)
+
+args = None
+def parse_args():
+	global args
+	global west_top
+	parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+			epilog=("This script supports XtensaTools but only when installed in a specific\n" +
+				"directory structure, example:\n" +
+				"myXtensa/\n" +
+				"└── install/\n" +
+				"       ├── builds/\n" +
+				"       │   ├── RD-2012.5{}/\n".format(xtensa_tools_version_postfix) +
+				"       │   │   └── Intel_HiFiEP/\n" +
+				"       │   └── RG-2017.8{}/\n".format(xtensa_tools_version_postfix) +
+				"       │       ├── LX4_langwell_audio_17_8/\n" +
+				"       │       └── X4H3I16w2D48w3a_2017_8/\n" +
+				"       └── tools/\n" +
+				"               ├── RD-2012.5{}/\n".format(xtensa_tools_version_postfix) +
+				"               │   └── XtensaTools/\n" +
+				"               └── RG-2017.8{}/\n".format(xtensa_tools_version_postfix) +
+				"                   └── XtensaTools/\n" +
+			"$XTENSA_TOOLS_ROOT=/path/to/myXtensa ...\n"), add_help=False)
+
+	parser.add_argument("-k", "--key", type=pathlib.Path, required=False,
+						help="Path to a non-default rimage signing key.")
+
+	parser.add_argument("-l", "--libraries", nargs="*", action=stores_libraries_arguments, default=[],
+						help=""" Function for CMake building modules.
+						We can build more then one module just by adding more module names.""")
+
+	parser.add_argument("-p", "--pack", required=False,
+						help="Create deployment zip headers pack from headers manifest")
+
+	args = parser.parse_args()
+
+	# print help message if no arguments provided
+	if len(sys.argv) == 1: #or args.help:
+		parser.epilog += "\nTo build module you must provide name and key to sign '"
+
+		parser.print_help()
+		sys.exit(0)
+
+
+def main():
+	parse_args()
+
+	if args.libraries:
+		cmake_build.build_libraries(LMDK_BUILD_DIR, RIMAGE_BUILD_DIR, args)
+	if args.pack:
+		header_pack.create_headers_pack()
```

Comment:
So the only thing the main script does is --help + some sanity checks?

Either put everything in a single script, or give the user two separate scripts and move the --help in some README.md file.

Reply:
BTW is most of this --help message even correct? I mean, is this funky toolchain layout requirement that was copied from xtensa-build-zephyr.py even still valid here?